src/axom/inlet/Field.hpp

```diff
@@ -87,9 +95,173 @@ class Field : public std::enable_shared_from_this<Field>
+  */
+  std::shared_ptr<Field> defaultValue(const char* value);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the default value of this Field.
+   *
+   * Set the default value for the Field in the input deck.
+   *
+   * \param [in] value The default boolean value
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> defaultValue(bool value);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the default value of this Field.
+   *
+   * Set the default value for the Field in the input deck.
+   *
+   * \param [in] value The default integer value
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> defaultValue(int value);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the default value of this Field.
+   *
+   * Set the default value for the Field in the input deck.
+   *
+   * \param [in] value The default double value
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> defaultValue(double value);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the range of this Field.
+   *
+   * Set the continuous range for the Field in the input deck.
+   *
+   * \param [in] startVal The start of the range
+   * 
+   * \param [in] endVal The end of the range
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> range(double startVal, double endVal);
+
+  /*!
+   *****************************************************************************
+   * \brief Set the range of this Field.
+   *
+   * Set the continuous range for the Field in the input deck.
+   *
+   * \param [in] startVal The start of the range
+   * 
+   * \param [in] endVal The end of the range
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> range(int startVal, int endVal);
+
+
+  /*!
+   *****************************************************************************
+   * \brief Set the valid values for this Field.
+   *
+   * \param [in] set An vector containing the set of allowed integer values
+   *
+   * \return Shared pointer to this Field instance
+   *****************************************************************************
+  */
+  std::shared_ptr<Field> validValues(std::vector<int> set);
```

Comment:
Change "std::vector<int> set" to "const std::vector<int>& set" so that it indicates that it isn't changing the vector but more importantly it doesn't copy the vector when this is called.

Reply:
I had originally attempted this, but it results in an error because copyBytes doesn't take a const pointer